cmake/FindGTest.cmake

```diff
@@ -34,6 +34,11 @@ if(NOT GTest_FOUND_SYSTEM)
 
     set(GTest_VERSION "1.10.0")
 
+    # supress all warning on m1 mac
+    if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")
+      set(CARGS -w)
```

Comment:
Let's only disable the specific warning that causes troubles and add a comment about why we are disabling that warning (if there is an issue on their issue tracker, we should link to it).